Assets/MixedRealityToolkit/Utilities/Editor/Setup/MixedRealityToolkitFiles.cs

```diff
@@ -144,25 +112,17 @@ static MixedRealityToolkitFiles()
 
         private static void SearchForFoldersAsync(string rootPath)
         {
-            foreach (MixedRealityToolkitModuleType module in Enum.GetValues(typeof(MixedRealityToolkitModuleType)))
+            foreach (string folder in Directory.EnumerateDirectories(rootPath))
             {
-                IEnumerable<string> directories = Directory.GetDirectories(rootPath, MixedRealityToolkitDirectory(module), SearchOption.AllDirectories);
-
-                if (directories.Count() == 0)
-                {
-                    directories = Directory.GetDirectories(rootPath, AlternateMixedRealityToolkitDirectory(module), SearchOption.AllDirectories);
-                }
-
-                directories = directories.Select(NormalizeSeparators);
-
-                foreach (string s in directories)
+                string normalizedFolder = NormalizeSeparators(folder);
+                if (FindMatchingModule(normalizedFolder, out MixedRealityToolkitModuleType module))
```

Comment:
I *think* that the code in this line up to 125 is shared with the 47-55 - could be worth sharing them.

Reply:
Correct, i added register/unregister functions for the common code here.